org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/XMLSchemaDiagnosticsTest.java

```diff
@@ -866,6 +876,35 @@ public void cvc_complex_type_2_3_elementBeforeText() throws BadLocationException
 		XMLAssert.testCodeActionsFor(xml, diagnostic, ca(diagnostic, te(3, 10, 3, 21, "")));
 	}
 
+	@Test
+	public void diagnosticRelatedInformationWithNoNamespaceSchemaLocation() throws Exception {
+		ContentModelSettings settings = new ContentModelSettings();
+		settings.setUseCache(true);
+		XMLValidationSettings validationSettings = new XMLValidationSettings();
+		validationSettings.setCapabilities(new PublishDiagnosticsCapabilities(true)); // with related information
+		settings.setValidation(validationSettings);
+
+		String xml = "<foo xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" \r\n" + //
+				"	 xsi:noNamespaceSchemaLocation=\"xsd/foo-invalid.xsd\">\r\n" + //
+				"	<bar />\r\n" + //
+				"</foo>";
+		Diagnostic diagnostic = new Diagnostic(r(1, 32, 1, 53), "There is '1' error in 'foo-invalid.xsd'.",
+				DiagnosticSeverity.Error, "xml");
+		diagnostic.setRelatedInformation(new ArrayList<>());
+		String xsdFileURI = getGrammarFileURI("foo-invalid.xsd");
+		diagnostic.getRelatedInformation().add(new DiagnosticRelatedInformation(l(xsdFileURI, r(1, 71, 1, 73)), ""));
+
+		XMLLanguageService xmlLanguageService = new XMLLanguageService();
+		// First validation
+		XMLAssert.testDiagnosticsFor(xmlLanguageService, xml, null, null, "src/test/resources/test.xml", false,
+				settings, //
+				diagnostic);
+		// Restart the validation to check the validation is working since Xerces cache the invalid XSD grammar
+		XMLAssert.testDiagnosticsFor(xmlLanguageService, xml, null, null, "src/test/resources/test.xml", false,
+				settings, //
+				diagnostic);
+	}
+

```

Comment:
I think you mentioned this already, but it would be good to also have a test with a `xsi:schemaLocation`

Reply:
fixed